docs/aws/audit/codebuildmonitoring/rules/codedeploy_auto_rollback_monitor_enabled.mdx

```diff
@@ -20,10 +20,271 @@ Performance Efficiency, Operational Excellence, Reliability, Security
+1. Log in to the AWS Management Console and navigate to the CodeBuild service.
+2. In the navigation pane, select "Build projects". This will display a list of all your build projects.
+3. Select the build project you want to check for CodeDeploy Auto Deploy and Monitoring. This will open the project details page.
+4. In the project details page, check the "Build details" section. If "Auto Deploy" and "Monitoring" are enabled, they will be listed here. If they are not listed, it means they are not enabled for this build project.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the resources.
+
+2. Once the AWS CLI is set up, you can list all the CodeBuild projects in your account using the following command:
+
+   ```
+   aws codebuild list-projects
+   ```
+
+   This command will return a list of all the CodeBuild projects in your account.
+
+3. For each project, you can describe the project to get more details about it. Use the following command:
+
+   ```
+   aws codebuild batch-get-projects --names project-name
+   ```
+
+   Replace 'project-name' with the name of your project. This command will return a JSON object with all the details about the project.
+
+4. In the returned JSON object, look for the 'triggers' section. If the 'webhook' field is set to 'true' and 'filterGroups' field contains 'EVENT: PUSH' and 'HEAD_REF: refs/heads/master', it means that Auto Deploy is enabled. If the 'cloudWatchLogs' field under 'logsConfig' is set to 'ENABLED', it means that Monitoring is enabled.
+
+   If these fields are not set as described, it means that Auto Deploy and Monitoring are not enabled in the CodeBuild project.
+</Accordion>
+
+<Accordion title='Using Python'>
+To check if CodeDeploy Auto Deploy and Monitoring is enabled in CodeBuild using Python scripts, you can use the AWS SDK for Python (Boto3). Here are the steps:
+
+1. **Setup AWS SDK for Python (Boto3):**
+   First, you need to set up Boto3. You can install it using pip:
+   ```
+   pip install boto3
+   ```
+   Then, configure your AWS credentials. You can do this by setting the following environment variables:
+   ```
+   AWS_ACCESS_KEY_ID = 'your_access_key'
+   AWS_SECRET_ACCESS_KEY = 'your_secret_key'
+   ```
+
+2. **List all CodeBuild projects:**
+   Use the `list_projects` method to get a list of all CodeBuild projects. Here is a sample script:
+   ```python
+   import boto3
+
+   client = boto3.client('codebuild')
+
+   response = client.list_projects()
+
+   for project in response['projects']:
+       print(project)
+   ```
+
+3. **Get project details:**
+   For each project, use the `batch_get_projects` method to get the project details. This will include information about the build configuration, including whether CodeDeploy is enabled. Here is a sample script:
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>